### PR TITLE
[android] Bump CompileSdkVersion to 34

### PR DESCRIPTION
**Pull request by rtsisyk**

### CODE REVIEW

**Comment by Jean-BaptisteC** [comment]
Please check #5663

**Comment by rtsisyk** [comment]
> Please check #5663

Thanks, @Jean-BaptisteC! It looks like that all three items in your list are new features that should be implemented and tested separately. Do you see any blocked for bumping SDK version? The sooner we bump the SDK version, the sooner we will find potential regressions.

**Comment by rtsisyk** [comment]
This is a blocker: https://github.com/organicmaps/organicmaps/pull/6199#issuecomment-1748525736

It looks like that permissions that I added to the manifest don't work as expected on Android 14. I will check later, don't have Android 14 emulator downloaded.

> Pixel 6 - Android 14
> 
>     * Start navigation
> 
> 
> ```
> FATAL EXCEPTION: main
> Process: app.organicmaps.beta, PID: 20467
> java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@5a98a6f with Intent { cmp=app.organicmaps.beta/app.organicmaps.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b45dff8 20467:app.organicmaps.beta/u0a615} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
> at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4839)
> at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
> at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
> at android.os.Handler.dispatchMessage(Handler.java:106)
> at android.os.Looper.loopOnce(Looper.java:205)
> at android.os.Looper.loop(Looper.java:294)
> at android.app.ActivityThread.main(ActivityThread.java:8177)
> at java.lang.reflect.Method.invoke(Native Method)
> at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
> at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
> Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b45dff8 20467:app.organicmaps.beta/u0a615} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
> at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
> at android.os.Parcel.createException(Parcel.java:3041)
> at android.os.Parcel.readException(Parcel.java:3024)
> at android.os.Parcel.readException(Parcel.java:2966)
> at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6761)
> at android.app.Service.startForeground(Service.java:775)
> at app.organicmaps.routing.NavigationService.onStartCommand(SourceFile:25)
> at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4821)
> at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0) 
> at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289) 
> at android.os.Handler.dispatchMessage(Handler.java:106) 
> at android.os.Looper.loopOnce(Looper.java:205) 
> at android.os.Looper.loop(Looper.java:294) 
> at android.app.ActivityThread.main(ActivityThread.java:8177) 
> at java.lang.reflect.Method.invoke(Native Method) 
> at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
> at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
> Caused by: android.os.RemoteException: Remote stack trace:
> at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2611)
> at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2322)
> at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1679)
> at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13265)
> at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3385)
> ```

This is surprising... We have that permission in the manifest... I can't reproduce on Android 13. Did you have some custom build?

**Comment by Jean-BaptisteC** [comment]
I use Android 14 :)

**Comment by biodranik** [comment]
Is it the only crash on Android 14 with this PR?

**Comment by Jean-BaptisteC** [comment]
Other crash:
- Select POI -> use button share
- Rotate device during share bottom sheet is visible -> crash

```
OMcore app.organicmaps.beta I organicmaps/Framework.cpp:397 PauseSurfaceRendering(): Pause surface rendering.
OMcore app.organicmaps.beta I organicmaps/Framework.cpp:295 DetachSurface(): Detach surface started. destroySurface = 1
OMcore app.organicmaps.beta I organicmaps/Framework.cpp:308 DetachSurface(): Destroy surface.
OMcore app.organicmaps.beta I organicmaps/Framework.cpp:330 DetachSurface(): Detach surface finished.
ActivityThread app.organicmaps.beta I Schedule relaunch activity: app.organicmaps.MwmActivity
gralloc4 app.organicmaps.beta E ERROR: Format allocation info not found for format: 3b
gralloc4 app.organicmaps.beta E ERROR: Format allocation info not found for format: 0
gralloc4 app.organicmaps.beta E Invalid base format! req_base_format = (<unrecognized format> 0x0), req_format = (<unrecognized format> 0x3b), type = 0x0
gralloc4 app.organicmaps.beta E ERROR: Unrecognized and/or unsupported format (<unrecognized format> 0x3b) and usage (CPU_READ_NEVER|CPU_WRITE_NEVER|GPU_TEXTURE|GPU_RENDER_TARGET|COMPOSER_OVERLAY 0xb00)
gralloc4 app.organicmaps.beta E ERROR: Format allocation info not found for format: 3b
gralloc4 app.organicmaps.beta E ERROR: Format allocation info not found for format: 0
gralloc4 app.organicmaps.beta E Invalid base format! req_base_format = (<unrecognized format> 0x0), req_format = (<unrecognized format> 0x3b), type = 0x0
gralloc4 app.organicmaps.beta E ERROR: Unrecognized and/or unsupported format (<unrecognized format> 0x3b) and usage (CPU_READ_NEVER|CPU_WRITE_NEVER|GPU_TEXTURE|GPU_RENDER_TARGET|COMPOSER_OVERLAY 0xb00)
OMcore app.organicmaps.beta I vulkan/android_vulkan_context_factory.cpp:413 ChangeSurface(): Surface changed 2400 1080
app.organi...s.location app.organicmaps.beta                 I  
OMcore app.organicmaps.beta I organicmaps/Framework.cpp:921 Java_app_organicmaps_Framework_nativeRemovePlacePageActivationListener(): Remove global map object listener
Surface app.organicmaps.beta E freeAllBuffers: 1 buffers were freed while being dequeued!
Surface app.organicmaps.beta E getSlotFromBufferLocked: unknown buffer: 0xb400007289c76630
vulkan app.organicmaps.beta E queueBuffer failed: Invalid argument (-22)
OMcore app.organicmaps.beta E vulkan/vulkan_base_context.cpp:495 Present(): CHECK(res == VK_SUCCESS) Vulkan error: vkQueuePresentKHR finished with code VK_ERROR_SURFACE_LOST_KHR
WindowOnBackDispatcher app.organicmaps.beta W sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@af46cc4
libc app.organicmaps.beta A Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 25098 (Thread-23), pid 24990 (rganicmaps.beta)
```

**Comment by biodranik** [comment]
Wow, does it reproduce on master?

**Comment by Jean-BaptisteC** [comment]
Yes
But not reproduce on 2023.09.22-27-Google

**Comment by rtsisyk** [comment]
Please extract this issue into a separate ticket - it is not related to SDK 34

**Comment by biodranik** [comment]
Crash in the beta with these commits on Android 34:

```
10-14 15:22:38.203 E/AndroidRuntime(13272): FATAL EXCEPTION: main
10-14 15:22:38.203 E/AndroidRuntime(13272): Process: app.organicmaps.beta, PID: 13272
10-14 15:22:38.203 E/AndroidRuntime(13272): java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@f79222d with Intent { cmp=app.organicmaps.beta/app.organicmaps.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{290e403 13272:app.organicmaps.beta/u0a413} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4838)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2288)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Handler.dispatchMessage(Handler.java:106)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Looper.loopOnce(Looper.java:205)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Looper.loop(Looper.java:294)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.ActivityThread.main(ActivityThread.java:8171)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at java.lang.reflect.Method.invoke(Native Method)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
10-14 15:22:38.203 E/AndroidRuntime(13272): Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{290e403 13272:app.organicmaps.beta/u0a413} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Parcel.createException(Parcel.java:3041)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Parcel.readException(Parcel.java:3024)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.os.Parcel.readException(Parcel.java:2966)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6761)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.Service.startForeground(Service.java:775)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at app.organicmaps.routing.NavigationService.onStartCommand(Unknown Source:63)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4820)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	... 9 more
10-14 15:22:38.203 E/AndroidRuntime(13272): Caused by: android.os.RemoteException: Remote stack trace:
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13343)
10-14 15:22:38.203 E/AndroidRuntime(13272): 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3385)
```

**Comment by Jean-BaptisteC** [comment]
More information here to fix crash -> https://developer.android.com/about/versions/14/changes/fgs-types-required#location

**Comment by biodranik** [comment]
@rtsisyk it should be easy to fix and test.

**Comment by rtsisyk** [comment]
The issue with "Starting FGS with type location" reported above has been fixed. I will wait for CI and merge.

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -872,9 +873,7 @@
         android:foregroundServiceType="location"
         android:exported="false"
         android:enabled="true"
-        android:stopWithTask="false"
-        android:permission="android.permission.ACCESS_FINE_LOCATION">
```
Comment (biodranik) [inline android/app/src/main/AndroidManifest.xml:876]: @rtsisyk is it a mistake?

**Pull request merged**